sktime/forecasting/tests/test_neuralforecast.py

```diff
@@ -225,29 +227,112 @@ def test_neural_forecast_with_auto_against_given_freq(model_class, freq) -> None
-    ):
-        # define model
-        model = model_class(freq="auto", max_steps=5, trainer_kwargs={"logger": False})
+    y = pandas.Series(data=range(len(index)), index=index)
 
-        # attempt train
-        model.fit(y, fh=[1, 2, 3, 4])
+    model = model_class(freq=1, max_steps=1, trainer_kwargs={"logger": False})
+    model_auto = model_class(freq="auto", max_steps=1, trainer_kwargs={"logger": False})
 
-    # should work with freq passed as param
-    model = model_class(freq="W", max_steps=5, trainer_kwargs={"logger": False})
+    model.fit(y, fh=[1, 2, 3])
+    model_auto.fit(y, fh=[1, 2, 3])
 
-    # attempt train
-    model.fit(y, fh=[1, 2, 3, 4])
+    pred = model.predict()
+    pred_auto = model_auto.predict()
+
+    # check prediction
+    pandas.testing.assert_series_equal(pred, pred_auto)
+
+
+@pytest.mark.parametrize(
+    "index",
+    [
+        pandas.RangeIndex(start=0, stop=20),
+        pandas.Index(range(20)),
+    ],
+)
+@pytest.mark.parametrize("model_class", [NeuralForecastLSTM, NeuralForecastRNN])
+@pytest.mark.skipif(
+    not run_test_for_class([NeuralForecastLSTM, NeuralForecastRNN]),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_neural_forecast_with_auto_freq_on_missing_int_like(index, model_class) -> None:
+    """Test with freq set to 'auto' on int-like index with missing values."""
+
+    index = index.drop(random.choices(index, k=5))
+    y = pandas.Series(data=range(len(index)), index=index)
+
+    model = model_class(freq="auto", max_steps=1, trainer_kwargs={"logger": False})
+
+    with pytest.raises(ValueError, match="could not interpret freq"):
+        model.fit(y, fh=[1, 2, 3])
+
+
+@pytest.mark.parametrize(
+    "index",
+    [
+        pandas.date_range(start="2024-01-01", periods=10),
+        pandas.period_range(start="2024-01-01", periods=10),
+    ],
+)
+@pytest.mark.parametrize("model_class", [NeuralForecastLSTM, NeuralForecastRNN])
+@pytest.mark.skipif(
+    not run_test_for_class([NeuralForecastLSTM, NeuralForecastRNN]),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_neural_forecast_with_auto_freq_on_date_like(index, model_class) -> None:
+    """Test with freq set to 'auto' on date-like index."""
+
+    y = pandas.Series(data=range(len(index)), index=index)
+
+    model = model_class(freq="D", max_steps=1, trainer_kwargs={"logger": False})
+    model_auto = model_class(freq="auto", max_steps=1, trainer_kwargs={"logger": False})
+
+    model.fit(y, fh=[1, 2, 3])
+    model_auto.fit(y, fh=[1, 2, 3])
+
+    pred = model.predict()
+    pred_auto = model_auto.predict()
+
+    # check prediction
+    pandas.testing.assert_series_equal(pred, pred_auto)
+
+
+@pytest.mark.parametrize(
+    "index",
+    [
+        pandas.date_range(start="2024-01-01", periods=10),
+        # freq is preserved in index even in missing data
+        # pandas.period_range(start='2024-01-01', periods=10),
```

Comment:
Can you please add one more list of manually created dates?

For example, `pandas.to_datetime(["2000-01-01", "2000-01-02", "2000-01-04", "2000-01-05"])`.

Reply:
Question: what happens with `pandas.PeriodIndex(data=["2000-01-01", "2000-01-02", "2000-01-04", "2000-01-05"],freq="D")`? Not just in this test, but with the auto behaviour itself?

Reply:
with or without auto, it works without any error
what seems to happen is that in `pandas.PeriodIndex` freq is reserved with each index, so even if there are missing values, `ForecastingHorizon` is able to infer it.